include/mbedtls/psa_util.h

```diff
@@ -124,35 +124,38 @@ static inline psa_key_usage_t mbedtls_psa_translate_cipher_operation(
 
 /* Translations for hashing. */
 
+/* Note: this function should not be used from inside the library, use
+ * mbedtls_hash_info_psa_from_md() from the internal hash_info.h instead.
+ * It is kept only for compatibility in case applications were using it. */
 static inline psa_algorithm_t mbedtls_psa_translate_md( mbedtls_md_type_t md_alg )
 {
     switch( md_alg )
     {
-#if defined(MBEDTLS_MD5_C)
```

Comment:
What is the purpose of this addition? To clearly signal dependencies?
Defining `MBEDTLS_MD5_C` will add `PSA_WANT_ALG_MD5` (without `MBEDTLS_PSA_CRYPTO_CONFIG`), and defining `PSA_WANT_ALG_MD5` will add `MBEDTLS_MD5_C` (with `MBEDTLS_PSA_CRYPTO_CONFIG` ) in `config_psa.h`.